hazelcast/src/main/java/com/hazelcast/spi/impl/SlowOperationLog.java

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright (c) 2008-2013, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.spi.impl;
+
+import com.eclipsesource.json.JsonArray;
+import com.eclipsesource.json.JsonObject;
+import com.eclipsesource.json.JsonValue;
+import com.hazelcast.management.JsonSerializable;
+import com.hazelcast.util.ConcurrencyUtil;
+import com.hazelcast.util.ConstructorFunction;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.TimeUnit;
+
+import static com.hazelcast.util.JsonUtil.getArray;
+import static com.hazelcast.util.JsonUtil.getInt;
+import static com.hazelcast.util.JsonUtil.getLong;
+import static com.hazelcast.util.JsonUtil.getString;
+
+public final class SlowOperationLog implements JsonSerializable {
+    private static final int SHORT_STACKTRACE_LENGTH = 200;
+
+    private static final ConstructorFunction<Integer, Invocation> INVOCATION_CONSTRUCTOR_FUNCTION
+            = new ConstructorFunction<Integer, Invocation>() {
+        @Override
+        public Invocation createNew(Integer arg) {
+            return new Invocation();
+        }
+    };
+
+    private final ConcurrentHashMap<Integer, Invocation> invocations = new ConcurrentHashMap<Integer, Invocation>();
+
+    private String operation;
+    private String stackTrace;
+    private int totalInvocations;
+
+    private transient String shortStackTrace;
+
+    public SlowOperationLog(String stackTrace, Object operation) {
+        this.operation = operation.toString();
+        this.stackTrace = stackTrace;
+        this.totalInvocations = 0;
+    }
+
+    String getOperation() {
+        return operation;
+    }
+
+    String getStackTrace() {
+        return stackTrace;
+    }
+
+    String getShortStackTrace() {
+        if (shortStackTrace == null) {
```

Comment:
I don't see that problem either. The original stack trace is always set in the constructor. So even if two threads would call that method, they would have the same data and would create the same value (just in different String instances). It should be the same behavior like `String.hashCode()` if called from multiple threads. What am I missing?